docs/reference/scripting.rst

```diff
@@ -1718,3 +1718,111 @@ BinaryFile.commit() : void
 BinaryFile.close() : void
     Closes the file. It is recommended to always call this function as soon as
     you are finished with the file.
+
+XmlFile
+~~~~~~~
+
+Inherits :ref:`script-xml-node`
+
+**Functions**
+
+new XmlFile(text : string)
+    Creates a new Xml document by parsing `text`.
+
+XmlFile.createNode(type : NodeType [, data : string = ""]) : XmlNode
+    Creates a new node attached to this document. The new node is not in the element
+    tree until it's been added somewhere. `data` is the initial value of XmlNode.text.
+    `type` cannot be NodeType.Root or NodeType.Other.
```

Comment:
I'd prefer to have separate `createElement`, `createTextNode` and `createCDATASection` functions for this. Why did you merge them into a single function with type parameter?